### PR TITLE
[R] Finalizes switch to markdown docu

### PR DESCRIPTION
**Pull request by mayer79**
Finalizes https://github.com/dmlc/xgboost/issues/9875 and solves https://github.com/dmlc/xgboost/issues/10329

### CODE REVIEW

File: R-package/R/xgb.Booster.R [hunk 1]
```diff
@@ -605,7 +604,7 @@ validate.features <- function(bst, newdata) {
 #' change the value of that parameter for a model.
 #' Use [xgb.parameters<-()] to set or change model parameters.
 #'
-#' The [xgb.attributes<-()] setter either updates the existing or adds one or several attributes,
+#' The \code{xgb.attributes<-} setter either updates the existing or adds one or several attributes,
```
Comment (trivialfis) [inline R-package/R/xgb.Booster.R:607]: Is this reversed?

File: R-package/R/xgb.cv.R [hunk 2]
```diff
@@ -2,141 +2,141 @@
...
-#'     \item \code{reg:squarederror} Regression with squared loss.
-#'     \item \code{binary:logistic} logistic regression for classification.
-#'     \item See \code{\link[=xgb.train]{xgb.train}()} for complete list of objectives.
-#'   }
-#'   \item \code{eta} step size of each boosting step
-#'   \item \code{max_depth} maximum depth of the tree
-#'   \item \code{nthread} number of thread used in training, if not set, all threads are used
-#' }
+#' @param params The list of parameters. The complete list of parameters is available in the
+#'   \href{http://xgboost.readthedocs.io/en/latest/parameter.html}{online documentation}.
```
Comment (trivialfis) [inline R-package/R/xgb.cv.R:6]: Do we still need the href?

File: R-package/R/xgb.Booster.R (hunk 1 above)
Comment (mayer79) [reply R-package/R/xgb.Booster.R:607]: Year. It did not look good in the rendered file.

File: R-package/R/xgb.cv.R (hunk 2 above)
Comment (mayer79) [reply R-package/R/xgb.cv.R:6]: Well spottet! Indeed, I can simplify it to [description](link).

File: R-package/R/xgb.Booster.R (hunk 1 above)
Comment (mayer79) [reply R-package/R/xgb.Booster.R]: Actually, the link points to the current page. This is why I wanted to remove the link. Now, I have replaced \code{} by ``

File: R-package/R/xgb.cv.R [hunk 3]
```diff
@@ -2,141 +2,141 @@
...
-#' @param callbacks a list of callback functions to perform various task during boosting.
-#'        See \code{\link{xgb.Callback}}. Some of the callbacks are automatically created depending on the
-#'        parameters' values. User can provide either existing or their own callback methods in order
-#'        to customize the training process.
-#' @param ... other parameters to pass to \code{params}.
+#'   This is not supported when `data` has `group` field.
+#' @param verbose Logical flag. Should statistics be printed during the process?
+#' @param print_every_n Print each nth iteration evaluation messages when `verbose > 0`.
+#'   Default is 1 which means all messages are printed. This parameter is passed to the
+#'   [xgb.cb.print.evaluation)] callback.
```
Comment (trivialfis) [inline R-package/R/xgb.cv.R:74]:
```suggestion
#'   [xgb.cb.print.evaluation()] callback.
```

**Review by trivialfis — APPROVED** [review]
Thank you for modernizing the R doc!

**Pull request merged**